File: Java/Migration/unit-tests-java/src/test/java/p01_core_runner/Ex01RememberingDataBetweenTests.java
package p01_core_runner;

import org.testng.annotations.Test;

//Questions:
//1. Which one runs first?
//2. Do both pass?
public class Ex01RememberingDataBetweenTests {
  private int _i = 0;

  @Test
  public void shouldIncrementANumberOneTime() {
    _i++;
    //assertEquals(_i, 1); //note the different order in the assertion!
  }

  @Test
  public void shouldAlsoIncrementANumberOneTime() {
    _i++;
    //assertEquals(_i, 1); //note the different order in the assertion!
  }
}

File: Java/Migration/unit-tests-csharp/unit-tests-csharp/P01CoreRunner/Ex01RememberingDataBetweenTests.cs
using NUnit.Framework;

namespace unit_tests_csharp.P01CoreRunner
{
  //Questions:
  //1. Which one runs first?
  //2. Do both pass?
  public class Ex01RememberingDataBetweenTests
  {
    private int _i = 0;

    [Test]
    public void ShouldIncrementANumberOneTime()
    {
      _i++;
      //Assert.AreEqual(1, _i); //TODO incomment
    }

    [Test]
    public void ShouldAlsoIncrementANumberOneTime()
    {
      _i++;
      //Assert.AreEqual(1, _i); //TODO incomment
    }
  }
}
